ghengin-core/ghengin-core/Ghengin/Core/Render/Pipeline.hs

```diff
@@ -135,7 +139,7 @@ makeRenderPipelineWith gps renderPass shaderPipeline props0 = Linear.do
 
   -- Make the resource map for this render pipeline using the dummyRP
   logT "Making resources"
-  (resources0, props1) <- makeResources props0
+  (resources0, props1) <- makeResources ((Ur (fromMaybe (error "Impossible1") (IM.lookup 0 descSetMap))) :: (Ur BindingsMap)) props0
```

Comment:
Here, we have descSetMap. As I understand it, it's a IntMap with the following integers:
```
This makes the descriptor set #0 bound once per pipeline
           the descriptor set #1 bound once per material
           the descriptor set #2 bound once per object
```

Since this is just making the pipeline resources, we can just select #0, right?

Maybe I'm getting this wrong.

Reply:
Here it's correct. Basically, `makeResources` should receive the appropriate `BindingsMap` and `RenderPipeline` should store the full reified `DescriptorSetMap`.

Whenever you call `makeResources` you have to lookup the right `BindingsMap`.